sql/log.cc

```diff
@@ -9090,8 +9081,26 @@ MYSQL_BIN_LOG::trx_group_commit_leader(group_commit_entry *leader,
       */
       DBUG_ASSERT(!cache_mngr->stmt_cache.empty() ||
                   !cache_mngr->trx_cache.empty()  ||
-                  current->thd->transaction->xid_state.is_explicit_XA() ||
-                  is_free_flush);
+                  current->thd->transaction->xid_state.is_explicit_XA());
```

Comment:
@knielsen, as your review mail mentions
```
The patch calls the trx_group_commit_leader()
```
I think you did not see this block in the latest commit that addresses (no more call of the func) this concern to my understanding without any major refactoring.